AdGuard Home

publicarray opened the pull request:
## Description

Adding AdGuard Home, It requires SSH access to give capabilities to the binary to bind to the privileged port (53) for DNS

Fixes #6069

I don't expect this package to be published in the current state with the current port 53 issue.

* Option 1- use a different port, but makes the package almost useless since for most client operating systems don't allow you to choose the DNS port to query
* Option 2 - somehow make it easier to add the capabilities for package updates. 
* Option 3 - I don't know, maybe Synology can be so kind to fix the Privilege Config: https://help.synology.com/developer-guide/privilege/privilege_config.html

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relevant tags.-->
- [ ] Bug fix
- [x] New Package
- [ ] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

publicarray committed:
AdGuardHome

publicarray committed:
update privilege file

publicarray committed:
Add Web UI to AdGuard Home and run DNS on port 5053
so it can run without root

publicarray committed:
improve icon

publicarray committed:
Patched out the privilege check as it wasn't working even with the correct capabilities.

publicarray committed:
make it easier to select the command

publicarray committed:
fix spelling

publicarray committed:
small fixes

publicarray committed:
extra word

publicarray commented:
@c0154936 since you opened the ticket can you try out this package? Download the test package from the actions tab at the top: https://github.com/SynoCommunity/spksrc/actions/runs/8763602582

c0154936 commented:
> @c0154936 since you opened the ticket can you try out this package? Download the test package from the actions tab at the top: https://github.com/SynoCommunity/spksrc/actions/runs/8763602582

I have manually installed the x64 package onto my Synology DS415+

I then ran the SSH command (but had to do it as ROOT):

setcap 'CAP_NET_BIND_SERVICE=+eip CAP_NET_RAW=+eip' /var/packages/adguardhome/target/bin/adguardhome

and then started the ADGUARDHOME software in the synology package center.

I can access the software now using the web interface (http://192.168.0.3:6053) and it presented me with the ADGUARDHOME welcome screen wizard. HOORAY!

I have set my windows laptop device to use ADGUARDHOME (192.168.0.3) as my DNS Server and my laptop device has appeared within the ADGuardHOME web interface as a client and it is also showing my all the DNS requests my laptop is making.

Looking good!

c0154936 commented:
I have not had a chance to test this yet but I have created a scheduled task on my synology NAS to run the following command when needed in future:

setcap 'CAP_NET_BIND_SERVICE=+eip CAP_NET_RAW=+eip' /var/packages/adguardhome/target/bin/adguardhome

![image](https://github.com/SynoCommunity/spksrc/assets/62215531/15af8d25-6ada-4b96-9dde-2fadc4f4d8bf)

![image](https://github.com/SynoCommunity/spksrc/assets/62215531/1567c5af-3656-456e-89fb-14169d9e546a)

publicarray commented:
Thanks for your feedback @c0154936 I thought about the scheduled tasks as well. Annoyingly this only works if you remember to run the task after an update, still better than ssh though

c0154936 commented:
Hi, just to keep you updated, I have updated ADGuardHOME within the web interface from v0.107.48 which you packaged to  v0.107.49.

![Pic1](https://github.com/SynoCommunity/spksrc/assets/62215531/dcccceda-118c-4ac1-9051-85a5f3d35de1)

After the update took effect, the software in the package center would not start back up. The status in the package center said "manually stopped". I did try to run it a few times but it kept going back to that status message.

So I ran the schedule task I created as mentioned in one of my above posts and the software is now running normally.

Everything is now working as normal (so far).

publicarray commented:
Thanks for the update @c0154936

> So I ran the schedule task I created as mentioned in one of my above posts and the software is now running normally.

This is expected an unfortunate as the permissions necessary to run the app are not officially supported by Synology. This command is a workaround to apply the permissions necessary and needs to run every time the binary is updated.

publicarray commented:
@michailf @hgy59 @th0ma7 

What do you think about packages that require root to be useful? I'm not sure if this package should be published. What do you guys think?  I do think AdGuardHome is very useful on a NAS, however some users may not get that it requires higher permissions to run.

I used @c0154936 screenshots above to start [a simple guide](https://github.com/SynoCommunity/spksrc/wiki/AdGuardHome) to help.

hgy59 commented:
> @michailf @hgy59 @th0ma7
> 
> What do you think about packages that require root to be useful? I'm not sure if this package should be published. What do you guys think? I do think AdGuardHome is very useful on a NAS, however some users may not get that it requires higher permissions to run.
> 
> I used @c0154936 screenshots above to start [a simple guide](https://github.com/SynoCommunity/spksrc/wiki/AdGuardHome) to help.

I use the patching of the privilege file in #6152 too.
My findings are, that after patching the privilege file, you need to "Repair" the package or start the service manually with `synopkg`. You can't start the package in the Package Center directly, until it has been started sucessfully once.

BTW 
_The privilege restrictions by Synology are boring. If you run a package as docker container you can officially run it as root..._

mreid-tt commented:
> @mreid-tt @hgy59 @th0ma7
> 
> What do you think about packages that require root to be useful? I'm not sure if this package should be published. What do you guys think? I do think AdGuardHome is very useful on a NAS, however some users may not get that it requires higher permissions to run.
> 
> I used @c0154936 screenshots above to start [a simple guide](https://github.com/SynoCommunity/spksrc/wiki/AdGuardHome) to help.

I've come to the same conclusion as @hgy59. These restrictions are forcing packages like this to be reviewed and signed by Synology for the official package repo. In the new [Developer Guide](https://global.download.synology.com/download/Document/Software/DeveloperGuide/Os/DSM/All/enu/DSM_Developer_Guide_7_enu.pdf) there is even a section which reads:

> **Install Development Token (For collaborative partners only)**
> If you are developing a package with root privilege, you are not able to install that package unless it is signed by synology. To deal with this security restriction, we provide a development token to bypass the signing restriction.

The challenge is that this development token is only valid for that NAS and installing the token to another NAS does not make the bypass work. All signs seem to point to requiring packages that need root to publish in Synology Package Center.

Now I've attempted to use the form at  https://www.synology.com/en-us/support/developer to query this issue since 2024-04-22 but beyond the initial _Synology Inquiry Ticket: #770842 (Developer)_ email acknowledgement, I've never received a response. This does not bode well for the overall process.

> BTW
> _The privilege restrictions by Synology are boring. If you run a package as docker container you can officially run it as root..._

This is indeed interesting and I've even seen a section of the new guide entitled _Compile Docker Package - Gitlab_ which goes into detail on how to compile a docker package _by using a well known version control opensource - Gitlab_.

What was also interesting is their brand new WIZARD_UIFILES [7.2.2] section which seems to change the whole mechanism to create wizard UI files. According to the document: _This new method involves using a render function to generate the wizard UI file. The render function is a Vue.js Framework structured function._ This will certainly be a fun adventure for future package creation.

mreid-tt commented:
@publicarray I don't know if this would help bit in the new [Developer Guide](https://global.download.synology.com/download/Document/Software/DeveloperGuide/Os/DSM/All/enu/DSM_Developer_Guide_7_enu.pdf) there is a recently added resource worker called _Systemd User Unit_.

I don't know if this would solve the issue since the function may still require `root` privileges to be setup but running the command that you need executed along with the updated manual through ChatGPT, it suggested the following implementation:

**File: conf/systemd/pkg-adguardhome/adguardhome.service:**
```
[Unit]
Description=AdGuard Home
After=network.target

[Service]
ExecStart=/var/packages/adguardhome/target/bin/adguardhome
Restart=on-failure
User=sc-adguardhome
Group=synocommunity
AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_NET_RAW
CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_NET_RAW
NoNewPrivileges=true

[Install]
WantedBy=multi-user.target
```

Much of this is beyond my current understanding but thought I'd look into any of the new resource workers in DSM 7.2 to see if there could be a solution there.

publicarray committed:
Merge branch 'SynoCommunity:master' into adguardhome

publicarray committed:
update adguardhome

fixes
 [error] parsing configuration file: unknown current schema version 29
due to updates that the older version can't read anymore preventing the app from starting

publicarray committed:
remove outdated patch

mreid-tt committed:
AdGuard Home: Update to v0.107.77, add privilege fix script and wizard
